### ISSUE TITLE
Server returned a status code of 500 while downloading https://poe.com/api/gql_POST

### ISSUE THREAD

**Issue by nextmine** [ading2210/poe-api#141]
Exception in thread Thread-6 (get_bot_thread):
Traceback (most recent call last):
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
Exception in thread     Exception in thread Thread-9 (get_bot_thread)self.run()
:
Thread-11 (get_bot_thread)  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
Traceback (most recent call last):
:
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    Traceback (most recent call last):
self._target(*self._args, **self._kwargs)  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner

  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
    self.run()
      File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
self.run()    chat_data = self.get_bot(bot["node"]["displayName"])

  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
        self._target(*self._args, **self._kwargs)
      self._target(*self._args, **self._kwargs)  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread

   File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
        chat_data = self.get_bot(bot["node"]["displayName"])    ^^^^^^
^^^chat_data = self.get_bot(bot["node"]["displayName"])
^                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^ ^^^^^^^^^^^^ 
^           ^^^^^^^^^^^^^^^^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
^^^^^^^^^^^^    ^^^^^^^^^^^^^^^^^^^^^^^^^^chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]^^^

  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
     File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                  chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]~~~~~~~~~~~~~      ~~~~~~~~         ~~~~~~~~~~~~
~~      ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^          ~~~~~~~~~~
KeyError: ~~~~'chatOfBotDisplayName'^^^^^^^^^^^^^^^^^^^^^^^^
~
~~~KeyError: 'chatOfBotDisplayName'
~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'chatOfBotDisplayName'
Exception in thread Thread-10 (get_bot_thread):
Traceback (most recent call last):
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
    chat_data = self.get_bot(bot["node"]["displayName"])
    Exception in thread      Thread-8 (get_bot_thread)       ^^^^^^^^^^^^^^^^^:
^^^^^^Traceback (most recent call last):
^^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
^^^^^^^^^^^    self.run()^^^^

  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                    self._target(*self._args, **self._kwargs)~
~  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
~~~~~~~~    chat_data = self.get_bot(bot["node"]["displayName"])~~
~    ~  ~~~~ Exception in thread ~ Thread-7 (get_bot_thread):
        Traceback (most recent call last):
~^^^^~  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
^~^~~^~^^^^^^    ^^^^self.run()^^^^
^^^^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
^^^^^^^^^^^^^^^^^^^^    ^^^^self._target(*self._args, **self._kwargs)^^
^^^^^^^
KeyError^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
: ^^^'chatOfBotDisplayName'^
^
chat_data = self.get_bot(bot["node"]["displayName"])
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
         chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
             ^^ ^^^^^^^^ ^Exception in thread  ^^^^^^^^^Thread-15 (get_bot_thread) ^:
 ^^^^^Traceback (most recent call last):
       ^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
~^^^^~^~^~~~~~~~~~~~~~~~~~^^^    ~^self.run()~~~^
~^^
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
^  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
^^^^^^    ^^    self._target(*self._args, **self._kwargs)
chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
^^
^ ^^                  chat_data = self.get_bot(bot["node"]["displayName"])^^^ ~^^^^^^^
~~~~~~~~~~~
 ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^KeyError
:       'chatOfBotDisplayName'KeyError  ^^^^^^^^^^:
^'chatOfBotDisplayName'^^
^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'chatOfBotDisplayName'
Exception in thread Thread-13 (get_bot_thread):
Traceback (most recent call last):
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
    chat_data = self.get_bot(bot["node"]["displayName"])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'chatOfBotDisplayName'
Exception in thread Thread-14 (get_bot_thread):
Traceback (most recent call last):
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
    chat_data = self.get_bot(bot["node"]["displayName"])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'chatOfBotDisplayName'
Exception in thread Thread-12 (get_bot_thread):
Traceback (most recent call last):
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 233, in get_bot_thread
    chat_data = self.get_bot(bot["node"]["displayName"])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\igovn\AppData\Local\Programs\Python\Python311\Lib\site-packages\poe.py", line 214, in get_bot
    chat_data = data["pageProps"]["data"]["chatOfBotDisplayName"]
                ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'chatOfBotDisplayName'
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (1/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (2/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (3/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (4/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (5/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (6/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (7/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (8/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (9/10)...
WARNING:root:Server returned a status code of 500 while downloading https://poe.com/api/gql_POST. Retrying (10/10)...




Code:
import poe

poe_token = 'token'
client = poe.Client(poe_token)

message = "Summarize the GNU GPL v3"
for chunk in client.send_message("capybara", message):
  pass
print(chunk["text"])

my ip is not banned, i can access poe.com.

**Comment by nextmine** [comment]
installed last version of library

**Comment by huyremy** [comment]
not yet fix, I have error too

**Comment by Bohaska** [comment]
Replacing ``chatOfBotDisplayName`` with ``chatOfBotHandle`` in this library's ``poe.py`` worked for me. Working again.

**Comment by ading2210** [comment]
This has been fixed in 0.4.8.
